src/mlpack/methods/ann/models/yolov3/yolov3_tiny_impl.hpp

```diff
@@ -0,0 +1,218 @@
+  model.Connect(maxPool11, convolution12);
+  model.Connect(convolution12, convolution13);
+
+  model.Connect(convolution13, convolution14);
+  model.Connect(convolution14, convolution15);
+  model.Connect(convolution15, detections16);
+
+  model.Connect(convolution13, convolution17);
+  model.Connect(convolution17, upsample18);
+
+  // Concat convolution8 + upsample18 => convolution19
+  model.Connect(upsample18, convolution19);
+  model.Connect(convolution8, convolution19);
+  // Set axis not necessary, since default is concat along channels.
+
+  model.Connect(convolution19, convolution20);
+  model.Connect(convolution20, detections21);
+  // Again, set axis not necessary, since default is concat along channels.
+
+  // Concatenation order shouldn't matter.
+  model.Connect(detections16, concatLayer22);
+  model.Connect(detections21, concatLayer22);
+
+  model.SetNetworkMode(false);
+  model.Reset();
+}
+
+template <typename OutputLayerType,
+          typename InitializationRuleType,
+          typename MatType>
+size_t YOLOv3Tiny<OutputLayerType, InitializationRuleType, MatType>
+::ConvolutionBlock(const size_t maps,
+                   const size_t kernel,
+                   const bool batchNorm,
+                   const Type reluSlope)
+{
+  if (kernel != 3 && kernel != 1)
+  {
+    std::ostringstream errMessage;
+    errMessage << "Kernel size for convolutions in yolov3-tiny must be 3"
+        "or 1, but you gave " << kernel << ".\n";
+    throw std::logic_error(errMessage.str());
+  }
+
+  size_t pad = kernel == 3 ? 1 : 0;
+  MultiLayer<MatType> block;
+  block.template Add<Convolution<MatType>>(
+    maps, kernel, kernel, 1, 1, pad, pad, "none", !batchNorm);
+
+  // set epsilon to zero, not used in darknet.
+  if (batchNorm)
+    block.template Add<BatchNorm<MatType>>(2, 2, 0, false);
+
+  block.template Add<LeakyReLU<MatType>>(reluSlope);
+  return model.Add(block);
+}
+
+template <typename OutputLayerType,
+          typename InitializationRuleType,
+          typename MatType>
+size_t YOLOv3Tiny<OutputLayerType, InitializationRuleType, MatType>
+::MaxPool2x2(const size_t stride)
+{
+  // All max pool layers have kernel size 2
+  MultiLayer<MatType> block;
+  if (stride == 1)
+  {
+    // One layer with odd input size, with kernel size 2, stride 1.
+    // Padding on the right and bottom are needed.
+    Type min = -arma::datum::inf;
+    block.template Add<Padding<MatType>>(0, 1, 0, 1, min);
+  }
+  block.template Add<MaxPooling<MatType>>(2, 2, stride, stride);
+  return model.Add(block);
+}
+
+template <typename OutputLayerType,
+          typename InitializationRuleType,
+          typename MatType>
+size_t YOLOv3Tiny<OutputLayerType, InitializationRuleType, MatType>
+::YOLO(const size_t imgSize,
+       const size_t gridSize,
+       const std::vector<Type>& anchors)
+{
+  return model.template Add<YOLOv3Layer<MatType>>(imgSize,
+    numAttributes, gridSize, predictionsPerCell, anchors);
```

Comment:
This function is so simple you could inline it, like you did when making the `NearestInterpolation` layer, but up to you.  It's just a thought I had.